### PR TITLE
[Breaking Change] Make emitter shapes easier to construct

### LINKED ISSUE #36: docs: `Default` is not implemented for EmitterShape
The docs say that `CircleSegment::opening_angle` will default to tau, but `EmitterShape` doesn't implement `Default`. That is, the following doesn't compile:

```rust
emitter_shape: EmitterShape::CircleSegment {
    radius: 0.0.,
    direction_angle: PI * 3. / 2.,
    ..default()
},
```

### PR DESCRIPTION
**Pull request by abnormalbrain**
Changes emitter shapes to use an inner struct to define properties, so that defaults can be used. Also defines a few convenience functions for creating emitter shapes, and adds `Default` to `EmitterShape` itself.

Closes #36

### CODE REVIEW

**Comment by johanhelsing** [comment]
Hey, sorry I never got back to you, and thanks for looking into this. I think my suggestion perhaps wasn't as good as I thought...

`EmitterShape::CircleSegment(CircleSegment {` etc. is quite of a mouthful. I think either a *bunch* of convenience methods are going to be needed, or some helper pattern to make it less tedious to write is needed.

One solution would perhaps be to implement `From<CircleSegment> for EmitterShape` or if it's possible `impl<T: ShapeTrait> From<T> for EmitterShape` if it's possible.

Then you could do:

```rust
emitter_shape: CircleSegment::default().into()
```

or 

```rust
emitter_shape: CircleSegment { /* values */ }.into()
```

**Comment by abnormalbrain** [comment]
Ahh, yeah fair point. I think implementing `From<>` for the segments would clean this up a good bit.

File: src/values.rs [hunk 1]
```diff
@@ -8,62 +8,143 @@ use bevy_transform::prelude::Transform;
 use rand::seq::SliceRandom;
 use rand::{prelude::ThreadRng, Rng};
 
+/// Describes an oriengted segment of a circle with a given radius.
```
Comment (johanhelsing) [inline src/values.rs:11]:
```suggestion
/// Describes an oriented segment of a circle with a given radius.
```

File: src/values.rs [hunk 2]
```diff
@@ -8,62 +8,143 @@ use bevy_transform::prelude::Transform;
...
+#[derive(Debug, Clone, Reflect, FromReflect)]
+pub struct CircleSegment {
+    /// The shape of the emitter, defined in radian.
```
Comment (johanhelsing) [inline src/values.rs:14]:
```suggestion
    /// The shape of the emitter, defined in radians.
```

repeats

File: src/values.rs [hunk 3]
```diff
@@ -8,62 +8,143 @@ use bevy_transform::prelude::Transform;
...
+    ///
+    /// The default is [`std::f32::consts::TAU`], which results particles going in all directions in a circle.
```
Comment (johanhelsing) [inline src/values.rs:16]:
```suggestion
    /// The default is [`std::f32::consts::TAU`], which results particles going in all directions in a circle.
```

perhaps it's better to say 2*PI, not everyone might know what tau is?

**Review by johanhelsing — APPROVED** [review]
I like this :)

File: src/values.rs (hunk 3 above)
Comment (abnormalbrain) [reply src/values.rs:16]:
```suggestion
    /// The default is `2 * PI`, which results particles going in all directions in a circle.
```

Ah yeah, fair point.

**Pull request merged**